src/FixedRankMatrices.jl

```diff
@@ -0,0 +1,219 @@
+using LinearAlgebra: diag, Diagonal, svd, SVD, rank, dot
+import Base: \, /, +, -, *
+@doc doc"""
+    FixedRankMatrices{M,N,K} <: Manifold
+
+The manifold of $m\times n$ real-valued matrices of fixed rank $k$, i.e.
+````math
+\mathcal M = \{ x \in \mathbb R^{m\times n} : \operatorname{rank}(x) = k \}.
+````
+
+# Representation with 3 matrix factors
+
+A point $x\in\mathcal M$ can be stored using orthonormal matrices
+$U\in\mathbb R^{m\times k}$, $V\in\mathbb R^{n\times k}$ as well as the $k$ singular
+values of $x = USV^\mathrm{T}$. In other words, $U$ and $V$ are from the manifolds
+[`Stiefel`](@ref)`(n,k)` and [`Stiefel`](@ref)`(m,k)`, respectively; see
+[`SVDMPoint`](@ref) for details
+
+The tangent space $T_x\mathcal M$ at a point $x\in\mathcal M$ with $x=USV^\mathrm{T}$
+is given by
+````math
+T_x\mathcal M = \bigl\{ UMV^\mathrm{T} + U_xV^\mathrm{T} + UV_x^\mathrm{T} : 
+    M \in \mathbb R^{k\times k},
+    U_x \in \mathbb R^{m\times k},
+    V_x \in \mathbb R^{n\times k}
+    \text{ s.t. }
+    U_x^\mathrm{T}U = 0_k,
+    V_x^\mathrm{T}V = 0_k
+\bigr\},
+````
+where $0_k$ is the $k\times k$ zero matrix. See [`UMVTVector`](@ref) for details.
+
+The (default) metric of this manifold is obtained by restricting the metric
+on $\mathbb R^{m\times n}$ to the tangent bundle.
+
+This representation follows
+> Bart Vandereycken: "Low-rank matrix completion by Riemannian Optimization,
+> SIAM Journal on Optimization, 23(2), pp. 1214–1236, 2013.
+> doi: [10.1137/110845768](https://doi.org/10.1137/110845768),
+> arXiv: [1209.3834](https://arxiv.org/abs/1209.3834).
+"""
+struct FixedRankMatrices{M,N,K,T} <: Manifold end
+FixedRankMatrices(m::Int, n::Int, k::Int, T::Type = Real) = FixedRankMatrices{m,n,k,T}()
```

Comment:
`Real` is an abstract type, which may be bad for performance. Maybe you need to introduce something like your own set of fields? As I was typing, I remembered related code from @Jutho

https://github.com/Jutho/TensorKit.jl/blob/ae270985e1d71bc982d912144d98ef8b560da9e2/src/spaces/vectorspaces.jl#L10-L19

So, depending on your type `T`, you may wish to put `ℝ` or `ℂ` as your last argument.

Reply:
While I like the notation that one gets there, the disadvantage is, that one can not “see” the `eltype(x)` of a point on a manifold from `M`, though I am also not 100% sure whether we need that somewhere explicitly.

Reply:
Maybe @mateuszbaran could help deciding, which way we want to go?

Reply:
Oh, you mean `T` is meant to encode the `eltype` of manifold points? That's fine, but then the default should not be `Real`, but, say, `Float64`. I thought this `T` is decoupled from `eltype`s, and just meant to encode whether the manifold is real or complex.

Reply:
Actually, I am not yet 100% sure, I had in mind to maybe use it that way. But you might also be right, that it might be enough to know whether it's Real or Complex, then we should of course follow your idea to provide field types.

Reply:
`T` was not supposed to encode `eltype` of points on a manifold, it was supposed to just be a tag (effectively equivalent to the code from `TensorKit.jl`, although using specific types makes the intent more clear). I think that generally manifold objects should not store a precise `eltype` of points.

BTW, `TensorKit.jl` looks interesting. Maybe it would make sense to integrate `Manifolds.jl` and `TensorKit.jl` at some level?

Reply:
Great. That's what I wanted to make sure :) So we mimic/adapt the types from `TensorKit.jl` in `Manifolds.jl` ?

Reply:
Yes, we can adapt the types from `TensorKit.jl` (though maybe without these single-letter Unicode aliases).

Reply:
Oh, I especially _like_ these... do you have a strongwomen opinion against that and why?

Reply:
It's not a strong opinion, just a suspicion that someone might want to use these symbols for something else and there will be incompatibility. You can use them if you like, we can remove them if they actually cause problems.

Reply:
You might be right. I will still start with them and we will follow your suggestion to remove them if they cause problems.

Reply:
Since also manifolds from other PRs are involved in this, I would like to postpone this to after finishing this PR and do a separate one for the type-change, see #62.

Reply:
OK, this can be done later.